ui-performance/index.md

```diff
@@ -0,0 +1,484 @@
+</li>
+
+<li markdown="1">
+In VS Code, bring up the command palette and enter "observatory".
+Select **Flutter: Open Observatory Timeline** from the list that pops up.
+If this command isn’t available, make sure that the app is running.
+{% comment %}
+[PENDING: Can you go to the timeline directly? If so, which command?]
+{% endcomment %}
+</li>
+</ol>
+
+#### Using Observatory's timeline
+
+<aside class="alert alert-info" markdown="1">
+**Note:** The Observatory UI and Flutter's custom timeline page are currently
+evolving.  For this reason, we aren't fully documenting the UI at this time.
+If you are comfortable experimenting with Observatory, and would like to give
+us feedback, please file [issues or feature
+requests](https://github.com/dart-lang/sdk/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-observatory)
+as you find them.
+</aside>
+
+### Identifying problems in the rasterizer graph
+
+Sometimes a scene results in a layer tree that is easy to construct,
+but expensive to render on the GPU thread. When this happens,
+the engine graph has no red, but the rasterizer graph shows red.
+In this case, you’ll need to figure out what your code is doing that is causing
+rendering code to be slow. Specific kinds of workloads are more difficult for
+the GPU.  They may involve unnecessary calls to
+[saveLayer](https://docs.flutter.io/flutter/dart-ui/Canvas/saveLayer.html),
+intersecting opacities with multiple objects, and clips or shadows in specific
+situations.
+
+If you suspect that the source of the slowness is during an animation, use the
+[timeDilation](https://docs.flutter.io/flutter/scheduler/timeDilation.html)
+property to greatly slow the animation down.
+
+You can also slow the animation speed using the Flutter Inspector.
+In the inspector's gear menu, select **Enable Slow Animations**.
+If you want more control of the animation speed, set the
+[timeDilation](https://docs.flutter.io/flutter/scheduler/timeDilation.html)
+property in your code.
+
+Is the slowness on the first frame, or on the whole animation?
+If it's the whole animation, is clipping causing the slow down?
+Maybe there's an
+alternative way of drawing the scene that doesn't use clipping. For example,
+overlay opaque corners onto a square instead of clipping to a rounded rectangle.
+If it's a static scene that's being faded, rotated, or otherwise manipluated,
+maybe a RepaintBoundary can help.
+
+#### Checking for offscreen layers
+
+The `saveLayer` method is one of the most expensive methods in the Flutter
+framework.  It’s useful when applying post-processing to the scene,
+but it can slow your app and should be avoided if you don’t need it.
+Even if you don’t call `saveLayer` explicitly, implicit calls may happen
+on your behalf. You can check whether your scene is using saveLayer with the
+[PerformanceOverlayLayer.checkerboardOffscreenLayers](https://docs.flutter.io/flutter/rendering/PerformanceOverlayLayer/checkerboardOffscreenLayers.html)
```

Comment:
@chinmaygarde, sounds like this toggle  is something we should do a better job of exposing in the tooling?